### DISCUSSION TITLE
how to create only one instance of Pocketbase() in flutter

### DISCUSSION THREAD

**Discussion by nateluk** [pocketbase/pocketbase#1044] [Q&A]
Hi, this might be more of a flutter architecture question but how would you go about creating only one instance of `Pocketbase()` client at the root of the app and be able to access it within any part of the app? (like an environment variable)

I'm a bit confused as to how to write this in flutter/dart - any pointers would be great!

**Comment by pchasco** [comment]
I would use either Provider or GetIt packages. Google usually recommends Provider for beginners at Flutter. The Provider package has excellent documentation, and you will find many tutorials online that will explain it bette than I can here.
